src/platform/common/utils/multiStepInput.ts

```diff
@@ -209,19 +208,19 @@ export class MultiStepInput<S> implements IMultiStepInput<S> {
                         const inputValue = input.value;
                         input.enabled = false;
                         input.busy = true;
-                        if (!(await validate(inputValue))) {
+                        const validationMessage = await validate(inputValue);
+                        if (!validationMessage) {
+                            input.validationMessage = '';
                             resolve(inputValue);
+                        } else {
+                            input.validationMessage = validationMessage;
```

Comment:
This wasn't being applied so if you hit enter and a failure occurs, nothing would show up.